src/extension/background-script/actions/nostr/signEventOrPrompt.ts

```diff
@@ -25,21 +25,33 @@ const signEventOrPrompt = async (message: MessageSignEvent, sender: Sender) => {
       };
     }
 
-    const hasPermission = await hasPermissionFor(
-      PermissionMethodNostr["NOSTR_SIGNMESSAGE"],
-      host
-    );
+    const hasPermission =
+      (await hasPermissionFor(
+        PermissionMethodNostr["NOSTR_SIGNMESSAGE"],
+        host
+      )) ||
+      (await hasPermissionFor(
+        PermissionMethodNostr["NOSTR_SIGNMESSAGE"] + "/" + event.kind,
+        host
+      ));
     if (!hasPermission) {
       const promptResponse = await utils.openPrompt<{
-        enabled: boolean;
         blocked: boolean;
+        permissionOption: string;
       }>({
         ...message,
         action: "public/nostr/confirmSignMessage",
       });
 
       // add permission to db only if user decided to always allow this request
-      if (promptResponse.data.enabled) {
+      if (promptResponse.data.permissionOption == "dont_ask_current") {
+        await addPermissionFor(
+          PermissionMethodNostr["NOSTR_SIGNMESSAGE"] + "/" + event.kind,
+          host
+        );
+      }
+
+      if (promptResponse.data.permissionOption == "dont_ask_any") {
```

Comment:
shouldn't we in JS always us `===` ? 
(even though it does not matter here. 

should those `dont_ask_any` `dont_ask_current` values be constants somewhere? - I think we have that constants file

Reply:
resolved